src/plugins/data/public/query/filter_manager/lib/compare_filters.ts

```diff
@@ -17,32 +17,53 @@
  * under the License.
  */
 
-import { defaults, isEqual, omit } from 'lodash';
+import { defaults, isEqual, omit, map, sortBy } from 'lodash';
 import { esFilters } from '../../../../common';
 
+/**
+ * Include disabled, negate and store when comparing filters
+ */
+export const COMPARE_FILTER_STATE = {
+  disabled: true,
+  negate: true,
+  state: true,
+};
+
+const mapFilter = (filter: esFilters.Filter, comparators: any, excludedAttributes: string[]) => {
+  const cleaned: esFilters.FilterMeta = omit(filter, excludedAttributes);
+
+  if (comparators.negate) cleaned.negate = filter.meta && Boolean(filter.meta.negate);
+  if (comparators.disabled) cleaned.disabled = filter.meta && Boolean(filter.meta.disabled);
+
+  return cleaned;
+};
+
+const mapFilterArray = (
+  filters: esFilters.Filter[],
+  comparators: any,
+  excludedAttributes: string[]
+) => {
+  return sortBy(
```

Comment:
Looks like either `sortBy` is redundant or 2nd argument (sorting comparator) is missing?